### PR TITLE
Spell out SLSQP in SLSQPLSQFitter class docstring

### LINKED ISSUE #9949: SLSQP should be spelt out in SLSQPLSQFitter docstring
The docstring for https://docs.astropy.org/en/stable/api/astropy.modeling.fitting.SLSQPLSQFitter.html currently says 'SLSQP optimization algorithm and least squares statistic' - but we should explicitly spell out SLSQP for anyone not familiar with it and preferably also link to some resources to explain more about it.

### PR DESCRIPTION
**Pull request by allardbrain**
### Description

This PR expands the `SLSQP` acronym to `Sequential Least Squares Programming (SLSQP)` for better clarity, and also adds a link to the corresponding documentation within the docstring as suggested by the issue requester.

Fixes #9949

### CODE REVIEW

**Comment by allardbrain** [comment]
Blerts, I forgot to add the `[skip travis]` bit at the end of the commit message. Do I need to redo that?

**Comment by pllim** [comment]
Don't worry about it. I'll cancel it.

**Comment by allardbrain** [comment]
Cool, thank you!

File: astropy/modeling/fitting.py [hunk 1]
```diff
@@ -969,14 +969,18 @@ def _wrap_deriv(params, model, weights, x, y, z=None):
...
+    least squares statistic.
 
     Raises
     ------
     ModelLinearityError
         A linear model is passed to a nonlinear fitter
 
+    Notes
+    ------
+    See https://docs.astropy.org/en/stable/api/astropy.modeling.optimizers.SLSQP.html
```
Comment (pllim) [inline astropy/modeling/fitting.py:982]: I think @astrofrog meant some external literature to explain the algorithm, not a link back to this package's documentation.

Comment (allardbrain) [reply astropy/modeling/fitting.py:982]: Hmm, OK. I'll see what I can find 👍

**Comment by allardbrain** [comment]
Oops, looks like I also left off the "Q" part of the acronym. I'll add that in as well.

File: astropy/modeling/fitting.py (hunk 1 above)
Comment (nden) [reply astropy/modeling/fitting.py:982]: The optimizer (linked to above) has a reference to the algorithm description and linking to it is how I read [this comment](https://github.com/astropy/astropy/issues/9949#issuecomment-586369218). I think it's OK to link to the optimizer in this package to pick up any specifics in setting up and calling the scipy algorithm.

Comment (pllim) [reply astropy/modeling/fitting.py:982]: In that case, please use the Sphinx `:ref:` linking, not hardcoding the URL. Thanks!

**Comment by allardbrain** [comment]
Hello again. I could use a nudge in the right direction on including `:ref:` in the Sphinx docs.

I've been referring to this section of the Sphinx reference documentation:

https://www.sphinx-doc.org/en/1.7/markup/inline.html#role-ref

I understand I need to replace my current code (`See <hard-coded url>`) with `See :ref: some-label`.

I also understand I need to define the label `some-label` in the section where I want the reference link to point to. The place I want the reference link to point to is this page in the documentation:

https://docs.astropy.org/en/stable/api/astropy.modeling.optimizers.SLSQP.html

However, as best I can tell, this HTML file is created automatically based on the docstring of this class:

https://github.com/astropy/astropy/blob/a0f5940bd00ff7259ed6b2391bfddecd007a2a43/astropy/modeling/optimizers.py#L96

I know I need to create the label `some-label` in restructured text, and that the label should go just before the thing I want to link to. However, I can't add restructured text in the Python file directly, so I'm not sure where to define the label in this case. Would it go in the class docstring somewhere?

Thanks in advance for any help you can provide!

**Comment by pllim** [comment]
Oooo, right. It is a class API... Sorry I didn't look too closely the first time. Well... I can see it go two ways:

1. Forget the `:ref:` and use `~astropy.modeling.optimizers.SLSQP`.
2. But a label above "References" in that class docstring and use it.

I am not 100% sure Option 2 will even work, so I'd say go with Option 1. Sorry again!

**Comment by embray** [comment]
I'll finish this up.  I think the change is clear.

**Review by embray — APPROVED** [review]
I fixed the issue with the reference.  This is good to go pending CI completion.

**Comment by embray** [comment]
Hmm, not really sure why there was a merge conflict there.

**Comment by astrofrog** [comment]
@embray these days we try and avoid including   master merged in pull requests, could you rebase to remove this?

**Comment by embray** [comment]
Yeah I always try to avoid that.  Not sure how I ended up with a merge commit in there.

**Comment by embray** [comment]
Rebased.  I should have added `[skip ci]` to the commit message while I was at it.  Oh well, I guess it's too late for the Python community to prevent global climate catastrophe anyways.

**Comment by embray** [comment]
I went ahead and cancelled the build on circleci but apparently I lack permissions to do that on travis.

**Comment by pllim** [comment]
Thanks, all!

**Pull request merged**